src/main/java/com/glovoapp/ownership/scanning/ParentPackageAnnotationScanner.java

```diff
@@ -1,23 +1,27 @@
-package com.glovoapp.ownership;
+package com.glovoapp.ownership.scanning;
 
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 
 import java.lang.annotation.Annotation;
 import java.lang.reflect.AnnotatedElement;
-import java.lang.reflect.Method;
 import java.util.Optional;
 
+@Slf4j
 @RequiredArgsConstructor
-public final class ParentPackageAnnotationScanner<A extends Annotation> {
+public class ParentPackageAnnotationScanner<A extends Annotation> implements AnnotationScanner<A> {
 
     private final Class<A> annotationClass;
 
+    @Override
     public Optional<A> scan(AnnotatedElement element){
 
-        String packageName = getPackageName(element).orElse("");
+        log.info("executing ParentPackageAnnotationScanner on element {}", element.toString());
+
+        String packageName = PackageScanningUtils.getPackageName(element).orElse("");
 
         while (packageName.length() > 0) {
-            packageName = getSuperPackageName(packageName);
+            packageName = PackageScanningUtils.getSuperPackageName(packageName);
 
             Class<?> packageInfo = null;
```

Comment:
[IDE] Please pay more attention to [your IDE warnings](https://www.jetbrains.com/help/idea/code-inspection.html). This code shows:

> Variable 'packageInfo' initializer 'null' is redundant 

```suggestion
            Class<?> packageInfo;
```

ᴛʜɪs ᴀᴄᴛɪᴏɴ ᴡᴀs ᴘᴀʀᴛɪᴀʟʟʏ ᴀᴜᴛᴏᴍᴀᴛᴇᴅ ʙʏ [GɪᴛHᴜʙ's ʀᴇsᴘᴏɴsᴇ ᴛᴇᴍᴘʟᴀᴛᴇs](https://github.com/settings/replies)